### PR TITLE
Feature/m2 781  library  activities+items

### PR DESCRIPTION
**Pull request by nmialik**
Снимок экрана 2023-03-14 в 12 01 52

### CODE REVIEW

File: src/modules/Library/features/Applet/Activity/Activity.styles.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { styled } from '@mui/system';
+import { Box, FormControlLabel } from '@mui/material';
+
+import { variables } from 'shared/styles/variables';
+import theme from 'shared/styles/theme';
```
Comment (kdanilenka32) [inline src/modules/Library/features/Applet/Activity/Activity.styles.ts:5]: change imports pls to 'shared/styles'

File: src/modules/Library/features/Applet/Activity/Activity.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+} from './Activity.styles';
+import { Item } from '../Item';
+
+export const Activity = ({
+  appletId,
+  setAddToBuilderDisabled,
+  activity: { id, name, items },
+}: ActivityProps) => {
+  const { watch, setValue } = useFormContext();
+  const watchItems = watch(items.map((item) => `${appletId}.${id}.${item.id}`));
```
Comment (kdanilenka32) [inline src/modules/Library/features/Applet/Activity/Activity.tsx:24]: pls change naming of watchItems

File: src/modules/Library/features/Applet/AppletImage/AppletImage.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { StyledTitleLarge } from 'shared/styles/styledComponents';
```
Comment (kdanilenka32) [inline src/modules/Library/features/Applet/AppletImage/AppletImage.tsx:1]: Should it be here? I mean folder

Comment (nmialik) [reply src/modules/Library/features/Applet/AppletImage/AppletImage.tsx]: yep, 
 while changing the folders structure, you moved to the shared the wrong component:)

but i'll fix it
these components need to be moved to the shared

Снимок экрана 2023-03-14 в 12 25 37
Снимок экрана 2023-03-14 в 12 25 48

File: src/modules/Library/features/Applet/Activity/Activity.styles.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  padding: ${theme.spacing(0.6, 2)};
+
+  &:not(:last-child) {
+    margin-bottom: ${theme.spacing(0.8)};
+  }
+`;
+
+export const StyledActivityHeader = styled(Box)`
+  display: grid;
+  grid-template-columns: 4.8rem auto;
```
Comment (alex-i-e) [inline src/modules/Library/features/Applet/Activity/Activity.styles.ts:21]: this style is the same as for the row 8, can be moved into a variable

File: src/modules/Library/features/Applet/Applet.styles.ts [hunk 5]
```diff
@@ -35,15 +42,66 @@ export const StyledButtonsContainer = styled(Box)`
   display: flex;
   justify-content: space-between;
 
-  .MuiButton-contained .MuiButton-startIcon {
-    svg {
-      fill: ${variables.palette.white};
+  .MuiButton-contained {
+    &:disabled .MuiButton-startIcon {
+      svg {
+        fill: ${variables.palette.contained_btn_disabled_text};
```
Comment (alex-i-e) [inline src/modules/Library/features/Applet/Applet.styles.ts:48]: should not these styles be applied in common styles for disabled buttons ?

File: src/modules/Library/features/Applet/Applet.tsx [hunk 6]
```diff
@@ -92,9 +118,11 @@ export const Applet = ({
               {t('viewDetails')}
             </Button>
             <Button
+              disabled={addToBuilderDisabled}
```
Comment (alex-i-e) [inline src/modules/Library/features/Applet/Applet.tsx:121]: `addToBuilderDisabled` supposed to be a boolean, maybe rename it to `isAddToBuilderDisabled` or  `shouldAddToBuilder` and use opposite value for it `!shouldAddToBuilder`?

File: src/modules/Library/features/Applet/Item/Item.styles.ts [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  .MuiFormControlLabel-root {
+    margin: 0;
+  }
+`;
+
+export const StyledItemHeader = styled(Box)`
+  display: grid;
+  grid-template-columns: 4.8rem auto;
```
Comment (alex-i-e) [inline src/modules/Library/features/Applet/Item/Item.styles.ts:24]: this style is the same as for the row 9, can be moved into a variable (the same styles in file `Activity.styles.ts`)

File: src/modules/Library/features/Applet/Item/Item.utils.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+  switch (item.responseType) {
+    case ItemResponseType.SingleSelection:
+    case ItemResponseType.MultipleSelection:
+      return (
+        <>
+          {item.options?.map(({ title, image }) => (
+            <StyledItemContentRow key={uniqueId()}>
+              <StyledItemSvg>{itemResponseTypes[item.responseType].icon}</StyledItemSvg>
+              {image && <StyledItemImage src={image} alt="Option image" />}
```
Comment (alex-i-e) [inline src/modules/Library/features/Applet/Item/Item.utils.tsx:21]: Should we use translation for `alt="Option image"`?

**Review by kdanilenka32 — APPROVED** [review]

File: src/modules/Library/features/Applet/Item/Item.utils.tsx (hunk 8 above)
Comment (nmialik) [reply src/modules/Library/features/Applet/Item/Item.utils.tsx:21]: hm, as I know, we don't translate alt

**Review by alex-i-e — APPROVED** [review]

**Pull request merged**